src/Authentication/Authentication/Cmdlets/InvokeGraphRequest.cs

```diff
@@ -118,65 +262,704 @@ protected override void ProcessRecord()
+
+            var cancellationToken = _cancelToken.Token;
+            var response = client.SendAsync(request, HttpCompletionOption.ResponseHeadersRead, cancellationToken)
+                .GetAwaiter()
+                .GetResult();
+            return response;
+        }
+
+        private long SetRequestContent(HttpRequestMessage request, IDictionary content)
+        {
+            if (request == null)
+                throw new ArgumentNullException(nameof(request));
+            if (content == null)
+                throw new ArgumentNullException(nameof(content));
+
+            var body = FormatDictionary(content);
+            return SetRequestContent(request, body);
+        }
+
+        private long SetRequestContent(HttpRequestMessage request, string content)
+        {
+            if (request == null)
+                throw new ArgumentNullException(nameof(request));
+
+            if (content == null)
+                return 0;
+
+            Encoding encoding = null;
+            if (ContentType != null)
+                // If Content-Type contains the encoding format (as CharSet), use this encoding format
+                // to encode the Body of the WebRequest sent to the server. Default Encoding format
+                // would be used if Charset is not supplied in the Content-Type property.
+                try
+                {
+                    var mediaTypeHeaderValue = MediaTypeHeaderValue.Parse(ContentType);
+                    if (!string.IsNullOrEmpty(mediaTypeHeaderValue.CharSet))
+                        encoding = Encoding.GetEncoding(mediaTypeHeaderValue.CharSet);
+                }
+                catch (FormatException ex)
+                {
+                    if (!SkipHeaderValidation)
+                    {
+                        var outerEx = new ValidationMetadataException("ContentTypeException", ex);
+                        var er = new ErrorRecord(outerEx, "WebCmdletContentTypeException",
+                            ErrorCategory.InvalidArgument, ContentType);
+                        ThrowTerminatingError(er);
+                    }
+                }
+                catch (ArgumentException ex)
+                {
+                    if (!SkipHeaderValidation)
+                    {
+                        var outerEx = new ValidationMetadataException("ContentTypeException", ex);
+                        var er = new ErrorRecord(outerEx, "WebCmdletContentTypeException",
+                            ErrorCategory.InvalidArgument, ContentType);
+                        ThrowTerminatingError(er);
+                    }
+                }
+
+            var bytes = StreamHelper.EncodeToBytes(content, encoding);
+            var byteArrayContent = new ByteArrayContent(bytes);
+            request.Content = byteArrayContent;
+
+            return byteArrayContent.Headers.ContentLength.Value;
+        }
+
+        private void FillRequestStream(HttpRequestMessage request)
```

Comment:
general comment: Add a summary to each method so we understand what we are about the read in the function body. The name is decent so I have an idea but this function body has over 100 lines in it so I have to look more than a glance to understand.